### PR TITLE
Stjs 830 selenium mock for buttonid

### PR DESCRIPTION
**Pull request by mrokosz-pgs**
*  added new feature
*  created step to open page with additional parameter

### CODE REVIEW

File: features/mock_for_button_id.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  Background:
+    Given JavaScript configuration is set for scenario based on scenario's @config tag
+    And User opens prepared payment form page WITH_ADDITIONAL_BUTTON
+
+  @base_config
+  Scenario: Click on button configured as button id
+    When User fills payment form with defined card VISA_CARD
+    And THREEDQUERY mock response is set to "ENROLLED_Y"
+    And User clicks Pay button - AUTH response is set to "OK"
```
Comment (msakowicz) [inline features/mock_for_button_id.feature:14]: You can also add this step: Then User will see payment status information: "Payment has been successfully processed" to be sure that payment has been processed correctly

File: features/mock_for_button_id.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    Then THREEDQUERY request was sent only once with correct data
+
+  @base_config
+  Scenario: Click on button configured as additional button
+    When User fills payment form with defined card VISA_CARD
+    And User clicks Additional button
+    Then AUTH and THREEDQUERY requests were not sent
```
Comment (msakowicz) [inline features/mock_for_button_id.feature:21]: You should add steps for mock AUTH and THREEDQUERY. Without mock data this step 'Then AUTH and THREEDQUERY requests were not sent' will not work correctly

File: features/mock_for_button_id.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  Background:
+    Given JavaScript configuration is set for scenario based on scenario's @config tag
+    And User opens prepared payment form page WITH_ADDITIONAL_BUTTON
+
+  @base_config
+  Scenario: Click on button configured as button id
+    When User fills payment form with defined card VISA_CARD
+    And THREEDQUERY mock response is set to "ENROLLED_Y"
+    And User clicks Pay button - AUTH response is set to "OK"
+    Then THREEDQUERY request was sent only once with correct data
```
Comment (msakowicz) [inline features/mock_for_button_id.feature:15]: You can also check AUTH request

**Review by msakowicz — CHANGES_REQUESTED** [review]

File: features/mock_for_button_id.feature (hunk 3 above)
Comment (mrokosz-pgs) [reply features/mock_for_button_id.feature]: done

File: features/mock_for_button_id.feature (hunk 2 above)
Comment (mrokosz-pgs) [reply features/mock_for_button_id.feature:24]: done

File: features/mock_for_button_id.feature (hunk 1 above)
Comment (mrokosz-pgs) [reply features/mock_for_button_id.feature:14]: dome

**Review by ngrzegorzewicz — APPROVED** [review]

**Review by ngrzegorzewicz — APPROVED** [review]

**Pull request merged**